Diary/EditorViewController/EditorViewController.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  EditorViewController.swift
+//  Diary
+//
+//  Created by JPush, Wonbi on 2022/12/21.
+//
+
+import UIKit
+
+final class EditorViewController: UIViewController {
+    private let editorView: EditorView = EditorView()
+    
+    override func loadView() {
+        super.loadView()
+        self.view = editorView
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillDisappear),
+            name: UIResponder.keyboardWillHideNotification,
+            object: nil
+        )
+        
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillAppear),
+            name: UIResponder.keyboardWillShowNotification,
+            object: nil
+        )
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        editorView.scrollToTop()
+    }
+    
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
```

Comment:
오, 이것을 하는 이유는 무엇인가요. 안 하면 어떻게되나요??

Reply:
EditorViewController가 해제되었을 때 notification이 해제되지 않아서 옵저버가 계속 불려지는 상황이 나올 수 있습니다!

Reply:
혹시 [removeObserver 공식문서](https://developer.apple.com/documentation/foundation/notificationcenter/1413994-removeobserver) 읽어보셨나용??👀

Reply:
읽어보니 9.0부터는 옵저버를 알아서 해제해주는군요! 이 부분에 대해 어느정도 알고는 있었지만, 좀 더 확실하게 명시해주고 싶어서 removeObserver메서드를 사용해 봤습니다. 다만, 이 메서드는 원하지 않는 옵저버까지 모두 해제할 위험이 있으므로, 결과적으로는 사용하지 않고 시스템이 처리하도록 로직을 수정하였습니다.
(공식문서에서도 `You shouldn't use this method to remove all observers from a long-lived object because your code may not be the only code adding observers that involve the object.` 라고 중요한 점! 이라며 강조하고 있네요!)